### PR TITLE
Feature/US-2.1-CreateRoles

### PR DESCRIPTION
**Pull request by Lonely-fr**

### CODE REVIEW

File: db/migrate/20220202121027_generate_default_user_roles.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class GenerateDefaultUserRoles < ActiveRecord::Migration[5.2]
+  def change
+    #create_general_roles
+    UserRole.create(name: "System admin")
```
Comment (e-popolamov) [inline db/migrate/20220202121027_generate_default_user_roles.rb:4]: move this part to seed.rb

File: app/controllers/user_roles_controller.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_user_role
+      @user_role = UserRole.find(params[:id])
+    end
+
+    # Only allow a trusted parameter "white list" through.
+    def user_role_params
+      params.fetch(:user_role, {})
```
Comment (e-popolamov) [inline app/controllers/user_roles_controller.rb:49]: params.permit bla-bla-bla

File: app/controllers/user_roles_controller.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    end
+  end
+
+  # DELETE /user_roles/1
+  def destroy
+    @user_role.destroy
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
```
Comment (e-popolamov) [inline app/controllers/user_roles_controller.rb:42]: this comments necessary?

File: app/controllers/user_roles_controller.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    def set_user_role
+      @user_role = UserRole.find(params[:id])
```
Comment (e-popolamov) [inline app/controllers/user_roles_controller.rb:44]: this part could raise ActiveRecord::RecordNotFound

File: app/controllers/user_roles_controller.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
+class UserRolesController < ApplicationController
+  before_action :set_user_role, only: [:show, :update, :destroy]
+
+  # GET /user_roles
+  def index
+    @user_roles = UserRole.all
+
+    render json: @user_roles
```
Comment (e-popolamov) [inline app/controllers/user_roles_controller.rb:8]:
```suggestion
    render json: UserRole.all
```

**Review by alllpim — APPROVED** [review]

**Review by Laranto-spb — APPROVED** [review]

**Review by StoreCarCharIsInNoSave — APPROVED** [review]

**Pull request merged**